src/libraries/System.Reflection.Emit/tests/PersistableAssemblyBuilder/AssemblySaveILGeneratorTests.cs

```diff
@@ -1486,5 +1585,591 @@ public void DeeperNestedTryCatchFilterFinallyBlocks()
+            GetCode(1 << 5); // n = 32 exp 128 was 68
+
+            static void GetCode(int num)
+            {
+                AssemblySaveTools.PopulateAssemblyBuilderTypeBuilderAndSaveMethod(out TypeBuilder type, out MethodInfo _);
+                MethodBuilder method = type.DefineMethod("meth1", MethodAttributes.Public | MethodAttributes.Static, typeof(int), Type.EmptyTypes);
+                var ilg = method.GetILGenerator();
+
+                var labels = new Label[num + 1];
+                for (int i = 0; i <= num; i++)
+                    labels[i] = ilg.DefineLabel();
+
+                ilg.Emit(OpCodes.Ldc_I4_0);
+                ilg.Emit(OpCodes.Br, labels[0]);
+
+                for (int i = num; --i >= 0;)
+                {
+                    ilg.MarkLabel(labels[i]);
+
+                    ilg.Emit(OpCodes.Ldc_I4_1);
+                    ilg.Emit(OpCodes.Ldc_I4_1);
+                    ilg.Emit(OpCodes.Ldc_I4_1);
+                    ilg.Emit(OpCodes.Ldc_I4_1);
+                    ilg.Emit(OpCodes.Add);
+                    ilg.Emit(OpCodes.Add);
+
+                    // Unconditional jump to "next" block (which is really before this code).
+                    ilg.Emit(OpCodes.Br, labels[i + 1]);
+                }
+
+                ilg.MarkLabel(labels[num]);
+
+                // Each block leaves two values on the stack. Add them into the previous value.
+                for (int i = 0; i < num; i++)
+                {
+                    ilg.Emit(OpCodes.Add);
+                    ilg.Emit(OpCodes.Add);
+                }
+
+                ilg.Emit(OpCodes.Ret);
+
+                type.CreateTypeInfo();
+                MethodInfo getMaxStackMethod = GetMaxStackMethod();
+                // The expected formula comes from (num - 1) * 2 + 1 + 5,
+                // ILs within the loop produces 2 adjustments for each, except 1st one
+                // last loop produce 2 + 1 adjustment because of the 1st instruction that loads 0
+                // the max stack for 1st loop is 5, 4 for the all other loops
+                Assert.Equal(num * 2 + 4, getMaxStackMethod.Invoke(ilg, null));
+            }
+        }
+
+        [Fact]
+        public void AmbiguousDepth()
+        {
+            GetCode();
+
+            static void GetCode()
+            {
+                AssemblySaveTools.PopulateAssemblyBuilderTypeBuilderAndSaveMethod(out TypeBuilder type, out MethodInfo _);
+                MethodBuilder method = type.DefineMethod("meth1", MethodAttributes.Public | MethodAttributes.Static, typeof(int), new[] { typeof(bool) });
+                var ilg = method.GetILGenerator();
+
+                // The label is targeted with stack depth zero.
+                var lab = ilg.DefineLabel();
+                ilg.Emit(OpCodes.Ldarg_0);
+                ilg.Emit(OpCodes.Brfalse, lab);
+
+                // The label is marked with a larger stack depth, one. This IL is invalid.
```

Comment:
So we don't detect \ throw on these cases, and CreateTypeInfo() doesn't either?

Reply:
These tests are imported from runtime ILGenerator tests https://github.com/dotnet/runtime/blob/9e57b6a280308b6e5dc85d1f0c9200cb698fb98b/src/libraries/System.Reflection.Emit.ILGeneration/tests/ILGenerator/Emit5Tests.cs#L250-L251
Personally, I am not sure this exact test makes sense, though runtime doesn't detect or throw ...